Pocket/pocket-ios
# Rewrote the ItemsListItemCell in SwiftUI and created components for all of the cell items.

## Summary
* Rewrite of ItemsListItemCell in SwiftUI
* Created reusable components and styles for ActionButton, Overflow Menu, Tags, Article Details, and more.

## References 
* [IN-1081: Write ItemListCell into SwiftUI for use in Search](https://getpocket.atlassian.net/browse/IN-1081?atlOrigin=eyJpIjoiMWExNWFiNzA2NTM0NDU2ODlhZWFlNDhjN2Y0YzAzZGQiLCJwIjoiaiJ9)

## Implementation Details
* Created ListItem and all dependent reusable components and styles
* Created ListItem+Constants so it is easy to update constants across the views
* I had to modify the ItemsListItemCell.Model by adding a dedicated swiftUITrackOverflow which conforms to ItemAction since the current trackOverflow conforms to UIAction which is not compatible with SwiftUI

## Test Steps
* The biggest area we need to test is the compatibility and design similarity between ListItem and ItemsListItemCell
* As we continue development of search we need to integrate the ListItem component to ensure that it functions as expected and meets the acceptance criteria for search 

## PR Checklist:

## Screenshots
<img width="565" alt="image" src="https://user-images.githubusercontent.com/8561013/206536193-fc0f17de-7fd7-44c3-a3e5-86fcbbacfab1.png">
<img width="565" alt="image" src="https://user-images.githubusercontent.com/8561013/206536270-b0f1e594-43e5-4415-a2c4-a7eadc432a4d.png">
